source/simulator/assembly.cc

```diff
@@ -1367,7 +1367,7 @@ namespace aspect
           Assert (false, ExcNotImplemented());
       }
 
-    const unsigned int block_idx = advection_field.block_index(introspection);
+    const unsigned int block_idx = advection_field.base_element(introspection);
```

Comment:
Did you run your PR? This needs to be ```block_index```! See where it is used? It accesses a matrix block using it.